axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/LegacyAxonServerEventStorageEngine.java

```diff
@@ -0,0 +1,254 @@
+        AggregateEventStream aggregateStream = connection.eventChannel().openAggregateStream(aggregateIdentifier);
+        return MessageStream.fromStream(aggregateStream.asStream(), this::convertToMessage, event -> Context.with(
+                                                                                                                    LegacyResources.AGGREGATE_IDENTIFIER_KEY,
+                                                                                                                    event.getAggregateIdentifier()).withResource(LegacyResources.AGGREGATE_TYPE_KEY,
+                                                                                                                                                                 event.getAggregateType())
+                                                                                                            .withResource(
+                                                                                                                    LegacyResources.AGGREGATE_SEQUENCE_NUMBER_KEY,
+                                                                                                                    event.getAggregateSequenceNumber()));
+    }
+
+    private EventMessage<byte[]> convertToMessage(Event event) {
+        return new GenericEventMessage<>(event.getMessageIdentifier(),
+                                         event.getPayload().getData().toByteArray(),
+                                         getMetaData(event.getMetaDataMap()),
+                                         Instant.ofEpochMilli(event.getTimestamp()));
+    }
+
+    private MetaData getMetaData(Map<String, MetaDataValue> metaDataMap) {
+        MetaData metaData = MetaData.emptyInstance();
+        for (Map.Entry<String, MetaDataValue> entry : metaDataMap.entrySet()) {
+            Object value = convertFromMetaDataValue(entry.getValue());
+            if (value != null) {
+                metaData = metaData.and(entry.getKey(), value);
+            }
+        }
+        return metaData;
+    }
+
+    public Object convertFromMetaDataValue(MetaDataValue value) {
+        switch (value.getDataCase()) {
+            case TEXT_VALUE:
+                return value.getTextValue();
+            case DOUBLE_VALUE:
+                return value.getDoubleValue();
+            case NUMBER_VALUE:
+                return value.getNumberValue();
+            case BOOLEAN_VALUE:
+                return value.getBooleanValue();
+            case DATA_NOT_SET, BYTES_VALUE:
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public MessageStream<EventMessage<?>> stream(@Nonnull StreamingCondition condition) {
+        TrackingToken trackingToken = condition.position();
+        if (trackingToken instanceof GlobalSequenceTrackingToken gtt) {
+            return new AxonServerMessageStream(connection.eventChannel().openStream(gtt.getGlobalIndex(), 32),
```

Comment:
We can't/shouldn't get rid of it, but having it (optionally) configurable would be better. Perhaps we should add a little config class for any optional configuration elements?

Reply:
Scrap that last comment. 
The contexts method is used to implement a check that the context exists after the purge. But since we call the purge API on Axon Server (previous versions deleted the context and recreated it), I wonder if all these methods are necessary.
On the other hand, if we decide to provide some APIs on the AxonServerTestContainer class, these could come in very handy again.